hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/s3guard/DumpS3GuardTable.java

```diff
@@ -0,0 +1,691 @@
+
+    return Pair.of(dirs, files);
+  }
+
+
+  /**
+   * Dump a single entry, and log it.
+   * @param csv CSV output file.
+   * @param md metadata to log.
+   */
+  private void dumpEntry(CsvFile csv, DDBPathMetadata md) {
+    LOG.info("{}", md.prettyPrint());
+    csv.entry(md);
+  }
+
+  /**
+   * Scan the metastore for all entries and dump them.
+   * There's no attempt to sort the output.
+   * @param csv file
+   * @return count of the number of entries.
+   */
+  private int scanMetastore(CsvFile csv) {
+    S3GuardTableAccess tableAccess = new S3GuardTableAccess(getStore());
+    ExpressionSpecBuilder builder = new ExpressionSpecBuilder();
+    Iterable<DDBPathMetadata> results = tableAccess.scanMetadata(
+        builder);
+    int count = 0;
+    for (DDBPathMetadata md : results) {
+      if (!(md instanceof S3GuardTableAccess.VersionMarker)) {
+        count++;
+        // print it
+        csv.entry(md);
+      }
+    }
+    return count;
+  }
+
+
+  /**
+   * Convert a timestamp in milliseconds to a human string.
+   * @param millis epoch time in millis
+   * @return a string for the CSV file.
+   */
+  private static String stringify(long millis) {
+    return new Date(millis).toString();
+  }
+
+  /**
+   * This is the JVM entry point for the service launcher.
+   *
+   * Converts the arguments to a list, then invokes {@link #serviceMain(List)}
+   * @param args command line arguments.
+   */
+  public static void main(String[] args) {
+    try {
+      serviceMain(Arrays.asList(args), new DumpS3GuardTable());
+    } catch (ExitUtil.ExitException e) {
+      ExitUtil.terminate(e);
+    }
+  }
+
+  /**
+   * The real main function, which takes the arguments as a list.
+   * Argument 0 MUST be the service classname
+   * @param argsList the list of arguments
+   * @param service service to launch.
+   */
+  static void serviceMain(
+      final List<String> argsList,
+      final AbstractS3GuardDiagnostic service) {
+    ServiceLauncher<Service> serviceLauncher =
+        new ServiceLauncher<>(service.getName());
+
+    ExitUtil.ExitException ex = serviceLauncher.launchService(
+        new Configuration(),
+        service,
+        argsList,
+        false,
+        true);
+    if (ex != null) {
+      throw ex;
+    }
+  }
+
+  /**
+   * Entry point to dump the metastore and s3 store world views
+   * <p>
+   * Both the FS and the store will be dumped: the store is scanned
+   * before and after the sequence to show what changes were made to
+   * the store during the list operation.
+   * @param fs fs to dump. If null a store must be provided.
+   * @param store store to dump (fallback to FS)
+   * @param conf configuration to use (fallback to fs)
+   * @param destFile base name of the output files.
+   * @param uri URI of store -only needed if FS is null.
+   * @throws ExitUtil.ExitException failure.
+   */
+  public static void dumpStore(
+      @Nullable final S3AFileSystem fs,
+      @Nullable DynamoDBMetadataStore store,
```

Comment:
Same thing here as in the other file: we have local implementation for the metadatastore. Please add `dynamo` to the classname, and i would be good to have these tools for the local store as well (in the future? separate work?).

Reply:
I will change the name on the hierarchy of classes to keep you happy